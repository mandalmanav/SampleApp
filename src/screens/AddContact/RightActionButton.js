import React, { useDebugValue } from 'react';
import { Pressable ,Alert} from 'react-native';
import { StyleSheet, Text, View, Button } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { addContactAction } from '../../store/actions/contacts';

export default ({navigation }) => {
    let {contact} = useSelector(state=>state)
    let dispatch = useDispatch()
    return (<Pressable onPress={async() => {
        if (contact.firstName==undefined||!contact.firstName.trim().length) {
                        Alert.alert(
                            "Invalid",
                            "First name cannot be left blank",
                            [
                                {
                                    text: "OK",
                                    onPress: () => {
                                        return
                                    },
                                    style: "cancel"
                                }
                            ]
                        );
                        return
                    }
                    if (contact.lastName==undefined||!contact.lastName.trim().length) {
                        Alert.alert(
                            "Invalid",
                            "Last name cannot be left blank",
                            [
                                {
                                    text: "OK",
                                    onPress: () => {
                                        return
                                    },
                                    style: "cancel"
                                }
                            ]
                        );
                        return
                    }
                    if (contact.mobile==undefined||String(contact.mobile).length < 10) {
                        Alert.alert(
                            "Invalid",
                            "Mobile number should consist of 10 digits",
                            [
                                {
                                    text: "OK",
                                    onPress: () => {
                                        return
                                    },
                                    style: "cancel"
                                }
                            ]
                        );
                        return
                    }
        await addContactAction(dispatch,contact)
        navigation.navigate('ContactDetails')
      }}><Text style={{ ...styles.primaryText, ...styles.actionFS, ...{ marginRight: 8 } }}>Save</Text>
      </Pressable>)
}

const styles = StyleSheet.create({
    primaryText: {
        color: '#4d90fe'
    },
    actionFS: {
        fontSize: 16
    }
});