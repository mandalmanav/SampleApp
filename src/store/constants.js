export const SET_CONTACTS = 'SET_CONTACTS'
export const ADD_CONTACT = 'ADD_CONTACT'
export const UPDATE_CONTACT = 'UPDATE_CONTACT'
export const DELETE_CONTACT = 'DELETE_CONTACT'
export const SET_SEARCH = 'SET_SEARCH'
export const SET_SELECTED = 'SET_SELECTED'
export const SET_UPDATE_OBJECT = 'SET_UPDATE_OBJECT'
export const SET_PROPERTY = 'SET_PROPERTY'