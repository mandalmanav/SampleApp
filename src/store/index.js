import { combineReducers } from "redux";
import { contactsReducer, updateObject } from './reducers/contactsReducers'
import { searchTermReducer } from './reducers/contactsReducers'
import { contactReducer } from "./reducers/contactsReducers"

export default combineReducers({
    contacts: contactsReducer,
    contact: contactReducer,
    searchTerm: searchTermReducer,
    updateObject:updateObject

})