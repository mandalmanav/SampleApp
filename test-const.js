export const CONTACTS=[{
    firstName:'Manav',
    lastName:'Mandal',
    image:'',
    mobile:'9999999999',
    id:'q',
    countryCode:'+91'
},{
    firstName:'Aman',
    lastName:'Vatsa',
    image:'',
    mobile:'9999999999',
    id:'r',
    countryCode:'+91'
}]

export const SEARCH_TERM='am'