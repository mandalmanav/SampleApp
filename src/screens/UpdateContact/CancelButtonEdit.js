import React, { useDebugValue } from 'react';
import { Pressable } from 'react-native';
import { StyleSheet, Text, View, Button } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { addContactAction, updateContactAction } from '../../store/actions/contacts';
import { SET_SELECTED } from '../../store/constants';

export default ({ navigation }) => {
    let { contact, contacts } = useSelector(state => state)
    let dispatch = useDispatch()
    return (<Pressable onPress={async () => {
        let oldContact = contacts.filter(c => c.id == contact.id)[0]
        dispatch({
            type: SET_SELECTED,
            payload: oldContact
        })

        navigation.navigate('ContactDetails')
    }}><Text style={{ ...styles.primaryText, ...styles.actionFS, ...{ marginLeft: 8 } }}>Cancel</Text>
    </Pressable>)
}

const styles = StyleSheet.create({
    primaryText: {
        color: '#4d90fe'
    },
    actionFS: {
        fontSize: 16
    }
});