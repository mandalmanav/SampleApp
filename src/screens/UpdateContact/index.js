
import React, { useState, useEffect } from 'react'
import { View, Text, StyleSheet, Image, TextInput, Pressable, Alert } from 'react-native'
import { Camera } from 'expo-camera';
import { addContactAction, updateContactAction } from '../../store/actions/contacts';
import { useDispatch, useSelector } from 'react-redux';
import { SET_CONTACTS, SET_PROPERTY, SET_SELECTED, SET_UPDATE_OBJECT, UPDATE_CONTACT } from '../../store/constants';
import { updateObject } from '../../store/reducers/contactsReducers';

export default ({ navigation, route }) => {
  
    let { contact } =useSelector(state => state)
    const [hasPermission, setHasPermission] = useState(null);
    const [type, setType] = useState(Camera.Constants.Type.back);
    let [openCamera, setOpenCamera] = useState(false)
    const dispatch = useDispatch()
    useEffect(() => {
        (async () => {
            const { status } = await Camera.requestPermissionsAsync();
            setHasPermission(status === 'granted');
        })();
    }, []);
    if (hasPermission === false) {
        return <Text>No access to camera</Text>;
    }
    const setContact = (property,value) =>{
        let temp = {...contact}
        temp[property]=value
        dispatch({
            type:SET_SELECTED,
            payload:temp
        })
    }
    const saveContact = () =>{
        alert(JSON.stringify(contact))
    }
    
    const snap = async () => {
        if (this.camera) {
            let photo = await this.camera.takePictureAsync();
            setContact('image',photo.uri)
            setOpenCamera(false)
        }
    };
    return <View style={styles.container}>
        {openCamera ? <Camera ref={ref => {
            this.camera = ref;
        }} style={styles.camera} type={type}>
            <View style={styles.buttonContainer}>
                <View>
                    <Pressable
                        style={styles.button}
                        onPress={() => {
                            setType(
                                type === Camera.Constants.Type.back
                                    ? Camera.Constants.Type.front
                                    : Camera.Constants.Type.back
                            );
                        }}>
                        <Text style={styles.text}> Flip </Text>

                    </Pressable>
                </View>
                <View>
                    <Pressable onPress={snap}>
                        <Text style={styles.text}>Capture</Text>
                    </Pressable>
                </View>
            </View>
        </Camera> : <View><View style={styles.detailsContainer}>

            <View style={styles.avatarContainer}>
                <View style={styles.avatarBox}>
                    
                    {
                        contact.image != null ?
                        <Pressable onPress={() => {
                            setOpenCamera(true)
                        }}>
                          
                          <Image
                            style={styles.avatar}
                            source={{
                                uri: contact.image,
                            }}
                        /> 
                         </Pressable>: <Pressable onPress={() => {
                            setOpenCamera(true)
                        }}>
                            <Text testID="add-image-button" style={{ textAlign: 'center' }}>Add Image</Text>
                        </Pressable>
                    }

                </View>
            </View>
            <View style={styles.details}>
                <TextInput testID="first-name" value={contact.firstName} onChangeText={(value)=>{
                    setContact('firstName',value)
                }} placeholder={'Enter First Name'}  style={styles.inputItem} />
                <TextInput testID="last-name" value={contact.lastName}  onChangeText={(value)=>{
                    setContact('lastName',value)
                }} placeholder={'Enter Last Name'}  style={styles.inputItem} />
            </View>

        </View>
            <View style={styles.mobileContainer}>
                <Text style={{ ...styles.actionFS, ...{ flex: 1, paddingLeft: 8 } }}>Mobile</Text>
                <TextInput testID="mobile"  onChangeText={(value)=>{
                    setContact('mobile',value)
                }}  keyboardType="number-pad" value={contact.mobile} maxLength={10}  style={{ ...styles.mobileInput, ...styles.inputItem }} placeholder="Add mobile"></TextInput>
            </View>
       
            </View>}
    </View>
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
        height: '100%',
        padding: 8
    },
    navBar: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    primaryText: {
        color: '#4d90fe'
    },
    actionFS: {
        fontSize: 18
    },
    avatarContainer: {
        flex: 1
    },
    avatar: {
        width: 50,
        height: 50,
        borderRadius: 50 / 2,
        overflow: "hidden",
    },
    avatarBox: {
        width: 70,
        height: 70,
        borderRadius: 70 / 2,
        overflow: "hidden",
        borderWidth: 1,
        borderColor: '#e5e5e5',
        alignItems: 'center',
        justifyContent: 'center'
    },
    detailsContainer: {
        flexDirection: 'row',
        paddingTop: 30
    },
    details: {
        paddingLeft: 20,
        flex: 3
    },
    inputItem: {
        borderBottomWidth: .5,
        borderBottomColor: '#e5e5e5',
        height: 40
    },
    mobileContainer: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    mobileInput: {
        flex: 3,
        marginLeft: 22
    },
    save: {
        borderWidth: 1,
        alignItems: 'center',
        width: 50,
        paddingTop: 6,
        paddingBottom: 6,
        paddingLeft: 6,
        paddingRight: 6
    },
    container: {
        flex: 1,
    },
    camera: {
        flex: 1,
    },
    buttonContainer: {
        flex: 1,
        backgroundColor: 'transparent',
        flexDirection: 'row',
        margin: 20,
        justifyContent: 'space-around'
    },
    button: {
        flex: 0.1,
        alignSelf: 'flex-end',
        alignItems: 'center',
    },
    text: {
        fontSize: 18,
        color: 'white',
    },
})
