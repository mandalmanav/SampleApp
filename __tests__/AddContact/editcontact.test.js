
import AddContact from '../../src/screens/AddContact'
import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { CONTACTS } from '../../test-const';
import { it } from '@jest/globals';
import { SET_SELECTED } from '../../src/store/constants';
import UpdateContact from '../../src/screens/UpdateContact';

const changedText = 'changedText'
describe('Test Edit Contact', () => {
    const expectedPayload1 = {
        type: SET_SELECTED, payload: {
           ...CONTACTS[0],firstName:changedText
        }
    }
    const expectedPayload2 = { type: SET_SELECTED, payload: {
        ...CONTACTS[0],lastName:changedText
    }}
    const expectedPayload3 = { type: SET_SELECTED, payload:  {
        ...CONTACTS[0],mobile:changedText
    } }

    let initialState = { contacts: CONTACTS, searchTerm: '', contact: CONTACTS[0] }
    const mockStore = configureStore()
    let store, wrapper
    store = mockStore(initialState)
    const changeAction = () => ({ type: SET_SELECTED })
    it('First name input box render and change text', () => {
        const { getByTestId } = render(<Provider store={store}><UpdateContact /></Provider>)
        let textBox = getByTestId('first-name')
        expect(textBox).not.toBeNull()
        expect(textBox.props.value).toBe(CONTACTS[0].firstName);
        fireEvent.changeText(textBox, changedText)
        const actions = store.getActions()
        expect(actions).toEqual([expectedPayload1])

    })
    it('Last name input box render and change text', () => {
        const { getByTestId } = render(<Provider store={store}><UpdateContact /></Provider>)
        let textBox = getByTestId('last-name')
        expect(textBox).not.toBeNull()
        expect(textBox.props.value).toBe(CONTACTS[0].lastName);
        fireEvent.changeText(textBox, changedText)
        const actions = store.getActions()
        expect(actions).toEqual([expectedPayload1,expectedPayload2])

    })
    it('Mobile number render and change correctly', () => {
        const { getByTestId } = render(<Provider store={store}><UpdateContact/></Provider>)
        let textBox = getByTestId('mobile')
        expect(textBox).not.toBeNull()
        expect(textBox.props.value).toBe(CONTACTS[0].mobile);
        fireEvent.changeText(textBox, changedText)
        const actions = store.getActions()
        expect(actions).toEqual([expectedPayload1,expectedPayload2,expectedPayload3])

    })
    // it('save button rendered and fire press event',()=>{
    //     const { getByTestId } = render(<Provider store={store}><AddContact/></Provider>)
    //     let button = getByTestId('save-button')
    //     expect(button.props.children).toBe('Save');
    //     expect(button).not.toBeNull()
    //    fireEvent.press(button)
    // })
    it('Match Snapshot', () => {
        const component = render(<Provider store={store}><UpdateContact /></Provider>)
        expect(component).toMatchSnapshot()
    })
})