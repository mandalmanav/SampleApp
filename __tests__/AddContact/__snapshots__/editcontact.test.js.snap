// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Edit Contact Match Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingTop": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#e5e5e5",
              "borderRadius": 35,
              "borderWidth": 1,
              "height": 70,
              "justifyContent": "center",
              "overflow": "hidden",
              "width": 70,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 50,
                  "overflow": "hidden",
                  "width": 50,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 3,
            "paddingLeft": 20,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Enter First Name"
          rejectResponderTermination={true}
          style={
            Object {
              "borderBottomColor": "#e5e5e5",
              "borderBottomWidth": 0.5,
              "height": 40,
            }
          }
          testID="first-name"
          underlineColorAndroid="transparent"
          value="Manav"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Enter Last Name"
          rejectResponderTermination={true}
          style={
            Object {
              "borderBottomColor": "#e5e5e5",
              "borderBottomWidth": 0.5,
              "height": 40,
            }
          }
          testID="last-name"
          underlineColorAndroid="transparent"
          value="Mandal"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 8,
          }
        }
      >
        Mobile
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        maxLength={10}
        onChangeText={[Function]}
        placeholder="Add mobile"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#e5e5e5",
            "borderBottomWidth": 0.5,
            "flex": 3,
            "height": 40,
            "marginLeft": 22,
          }
        }
        testID="mobile"
        underlineColorAndroid="transparent"
        value="9999999999"
      />
    </View>
  </View>
</View>
`;
